Quotient/events/roommessageevent.cpp

```diff
@@ -179,7 +176,34 @@ QMimeType RoomMessageEvent::mimeType() const
 {
     static const auto PlainTextMimeType =
         QMimeDatabase().mimeTypeForName("text/plain"_L1);
-    return _content ? _content->type() : PlainTextMimeType;
+    return content() ? content()->type() : PlainTextMimeType;
+}
+
+std::unique_ptr<EventContent::TypedBase> RoomMessageEvent::content() const
+{
+    const QJsonObject content = contentJson();
+    if (content.contains(MsgTypeKey) && content.contains(BodyKey)) {
+        auto msgtype = content[MsgTypeKey].toString();
+        bool msgTypeFound = false;
+        std::unique_ptr<EventContent::TypedBase> eventContent = nullptr;
+        for (const auto& mt : msgTypes) {
+            if (mt.matrixType == msgtype) {
+                eventContent = mt.maker(content);
+                msgTypeFound = true;
+            }
+        }
+
+        if (!msgTypeFound) {
+            qCWarning(EVENTS) << "RoomMessageEvent: unknown msg_type,"
+                              << " full content dump follows";
+            qCWarning(EVENTS) << formatJson << content;
+        }
+        return eventContent;
+    } else {
+        qCWarning(EVENTS) << "No body or msgtype in room message event";
+        qCWarning(EVENTS) << formatJson << fullJson();
+        return nullptr;
+    }
```

Comment:
While we are at this code...
```suggestion
    if (!content.contains(MsgTypeKey) || !content.contains(BodyKey)) {
        qCWarning(EVENTS) << "No body or msgtype in room message event";
        qCWarning(EVENTS) << formatJson << obj;
        return {};
    }
    if (auto it = std::ranges::find(msgTypes, content[MsgTypeKey].toString(),
                                    &MsgTypeDesc::matrixType);
        it != msgTypes.cend())
        return it->maker(content);

    qCWarning(EVENTS) << "RoomMessageEvent: unknown msgtype, full content dump follows";
    qCWarning(EVENTS) << formatJson << content;
    return {};
```